Sources/_StringProcessing/Unicode/ASCII.swift

```diff
@@ -0,0 +1,169 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2021-2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+
+private var _lineFeed: UInt8 { 0x0A }
+private var _carriageReturn: UInt8 { 0x0D }
+private var _lineTab: UInt8 { 0x0B }
+private var _formFeed: UInt8 { 0x0C }
+private var _space: UInt8 { 0x20 }
+private var _tab: UInt8 { 0x09 }
+
+private var _0: UInt8 { 0x30 }
+private var _9: UInt8 { 0x39 }
+private func _isASCIINumber(_ x: UInt8) -> Bool {
+  return (_0..._9).contains(x)
+}
+
+private var _a: UInt8 { 0x61 }
+private var _z: UInt8 { 0x7A }
+private var _A: UInt8 { 0x41 }
+private var _Z: UInt8 { 0x5A }
+
+private var _underscore: UInt8 { 0x5F }
+
+extension UInt8 {
+  var _isASCII: Bool { self < 0x80 }
+
+  // TODO: Bitvectors for the below
+
+  /// Assuming we're ASCII, whether we match `\d`
+  var _asciiIsDigit: Bool {
+    assert(_isASCII)
+    return(_0..._9).contains(self)
+  }
+
+  /// Assuming we're ASCII, whether we match `\h`
+  var _asciiIsHorizontalWhitespace: Bool {
+    assert(_isASCII)
+    return self == _space || self == _tab
+  }
+
+  /// Assuming we're ASCII, whether we match `\v`
+  var _asciiIsVerticalWhitespace: Bool {
+    assert(_isASCII)
+    switch self {
+    case _lineFeed, _carriageReturn, _lineTab, _formFeed:
+      return true
+    default:
+      return false
+    }
+  }
+
+  /// Assuming we're ASCII, whether we match `\s`
+  var _asciiIsWhitespace: Bool {
+    assert(_isASCII)
+    switch self {
+    case _space, _tab, _lineFeed, _lineTab, _formFeed, _carriageReturn:
+      return true
+    default:
+      return false
+    }
+  }
+
+  /// Assuming we're ASCII, whether we match `[a-zA-Z]`
+  var _asciiIsLetter: Bool {
+    assert(_isASCII)
+    return (_a..._z).contains(self) || (_A..._Z).contains(self)
```

Comment:
Check codgen for this; it should be just an AND or OR or BIC, followed by a subtract and compare. If not, we might need to rewrite it that way explicitly.

Reply:
Would that be better than a bit vector? 

These improvements are low priority as we also have such overwhelming inefficiencies elsewhere that time spent inside `_quickMatch` is close to negligible. However, I will definitely file something to track future improvements here.

Reply:
Yes, better than a bit vector.